cloudfront/wellcomelibrary.org/edge-lambda/src/api.ts

```diff
@@ -0,0 +1,59 @@
+import axios from "axios";
+import {Work, CatalogueResultsList} from "./catalogue";
+
+export type GetWorkResult = Work | Error
+
+const apiBasePath = 'https://api.wellcomecollection.org/catalogue/v2'
+
+type ApiQuery = {
+    query: string
+    include: Array<string>
+}
+
+async function* apiQuery(query: ApiQuery): AsyncGenerator<Work, void, void> {
+    const url = apiBasePath + '/works';
+    const apiResult = await axios.get(url, {params: query}  );
+
+    const resultList = apiResult.data as CatalogueResultsList
+    // const nextPage = resultList.nextPage
+    //
+    // if(resultList.results.length == 0) {
+    //     return Error("No matching Catalogue API results found")
+    // }
+
+    for(let i = 0; i < resultList.results.length; i++) {
+        yield resultList.results[i];
```

Comment:
Have you read this?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

Reply:
I did have a look over async generators/iterables, but got in a muddle and decided to come back later. I'm not sure how best to handle providing a generator that does async requests at the end of each page. I also want to be able to filter the output of that generator.